linkerd/app/core/src/metrics.rs

```diff
@@ -394,14 +394,18 @@ impl FmtLabels for RouteAuthzLabels {
 impl FmtLabels for HTTPLocalRateLimitLabels {
     fn fmt_labels(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         self.server.fmt_labels(f)?;
-        write!(
-            f,
-            ",ratelimit_group=\"{}\",ratelimit_kind=\"{}\",ratelimit_name=\"{}\",ratelimit_error=\"{}\"",
-            self.rate_limit.group(),
-            self.rate_limit.kind(),
-            self.rate_limit.name(),
-            self.error,
-        )
+        if let Some(rl) = &self.rate_limit {
+            write!(
+                f,
+                ",ratelimit_group=\"{}\",ratelimit_kind=\"{}\",ratelimit_name=\"{}\",ratelimit_error=\"{}\"",
+                rl.group(),
+                rl.kind(),
+                rl.name(),
+                self.error,
+            )
+        } else {
+            Ok(())
+        }
```

Comment:
```suggestion
        } else {
            write!(f, ",ratelimit_error=\"{}\"", self.error)
        }
```

one last tweak; in the case where no metadata exists, we can still format the error label